lib/model/content.dart

```diff
@@ -1599,6 +1584,30 @@ class _ZulipContentParser {
     for (final node in nodes) {
       if (node is dom.Text && (node.text == '\n')) continue;
 
+      // Oddly, the way a math block gets encoded in Zulip HTML is inside a <p>.
+      if (node case dom.Element(localName: 'p', className: '', nodes: [
+            dom.Element(
+              localName: 'span',
+              className: 'katex-display') && final child, ...])) {
+        if (node.nodes case [_]
+                         || [_, dom.Element(localName: 'br'),
+                                dom.Text(text: "\n")]) {
+          // This might be too specific; we'll find out when we do #190.
+          // The case with the `<br>\n` can happen when at the end of a quote;
+          // it seems like a glitch in the server's Markdown processing,
+          // so hopefully there just aren't any further such glitches.
+          if (currentParagraph.isNotEmpty) consumeParagraph();
+          if (imageNodes.isNotEmpty) consumeImageNodes();
+          result.add(parseMathBlock(child));
+          continue;
+        } else {
+          if (currentParagraph.isNotEmpty) consumeParagraph();
+          if (imageNodes.isNotEmpty) consumeImageNodes();
+          result.add(UnimplementedBlockContentNode(htmlNode: node));
+          continue;
```

Comment:
I believe this first commit is not quite NFC because the behavior changes in this case. In the old code, I believe this would end up with a ParagraphNode containing an UnimplementedInlineContentNode (and maybe other nodes), instead of with this one UnimplementedBlockContentNode.

That's the non-NFC-ness I was expecting at https://github.com/zulip/zulip-flutter/pull/1156#pullrequestreview-2613718416 . The change is fine; just the commit message should be adjusted: remove the NFC tag, and have the body explain that it's NFC except for this particular behavior change.

(I did work through the logic just now and satisfy myself that there's no *other* behavior change in this commit. So this splitting of commits was helpful, thanks.)